common/lib/dependabot/metadata_finders/base/changelog_finder.rb

```diff
@@ -314,24 +314,29 @@ def fetch_gitlab_file_list
         end
 
         def new_version
-          @new_version ||= git_source? ? new_ref : dependency.version
-          @new_version&.gsub(/^v/, "")
+          return @new_version if defined?(@new_version)
+
+          new_version = git_source? && new_ref ? new_ref : dependency.version
+          @new_version = new_version&.gsub(/^v/, "")
```

Comment:
Looks like the previous cached call wasn't doing `@new_version&.gsub(/^v/, "")` in any subsequent method calls